### DISCUSSION TITLE
Evoked activity?

### DISCUSSION THREAD

**Discussion by stevejryan** [flatironinstitute/nemos#100] [Q&A] [answered]
This is maybe a silly question, but would these GLM models work for activity that's evoked? Like, if there's a stimulus protocol where current steps are injected at different amplitudes?

**Comment by billbrod** [comment, marked as answer]
Yes, they do! We currently have a [tutorial about this](https://nemos-workshop-feb-2024.readthedocs.io/en/latest/generated/gallery/01_current_injection/) on the website from our February 2024 workshop, which we'll be integrating into this repo soon. Is that what you meant?

 (Note that if you want to try and follow along with that tutorial, the version used for the workshop uses a different version of the `GLM` object API, where the `X` / predictor and `y` / target inputs to `GLM.fit` are 2d and 1d, respectively, while the current `main` requires them to be 3d and 2d (the extra dimension is for neurons). We're probably going to integrate that API version, but you'll need to add an extra dimension for now with an extra `predictor = np.expand_dims(predictor, 1)` and `count = np.expand_dims(count, 1)` before passing them to `.fit`)

**Reply by stevejryan** [reply]
Oooh I haven't read it in extreme detail yet, but that looks like exactly what I'm trying to do! I'll try to follow along in the tutorial when I have a moment and report back :) 

Great meeting you at the neuro-rse shindig this afternoon :)

**Reply by billbrod** [reply]
Thanks Steve, it was great to meet you too!
